File: frontend/app/services/batch.ts
import api from "./api";

export interface BatchProcessItem {
  inspection_order_id: number;
  order_no: string;
  target_status: string;
  opinion?: string;
  signature?: string;
  current_version?: number;
}

export interface BatchProcessRequest {
  items: BatchProcessItem[];
  request_id?: string;
  operation: string;
}

export interface BatchItemResult {
  inspection_order_id: number;
  order_no: string;
  success: boolean;
  error_code: string | null;
  error_message: string | null;
  previous_status: string | null;
  current_status: string | null;
  suggestion: string | null;
  next_step: string | null;
}

export interface BatchStatistics {
  total_count: number;
  success_count: number;
  failed_count: number;
  skipped_count: number;
}

export interface BatchProcessResult {
  success: boolean;
  message: string;
  statistics: BatchStatistics;
  results: BatchItemResult[];
  request_id: string | null;
  completed_at: string;
}

export async function batchProcess(data: BatchProcessRequest): Promise<BatchProcessResult> {
  const response = await api.post<BatchProcessResult>("/api/batch/process", data);
  return response.data;
}

File: backend/schemas/batch.py
from pydantic import BaseModel, Field
from typing import Optional, List, Dict, Any
from datetime import datetime
from models.inspection import InspectionStatus


class BatchProcessItem(BaseModel):
    inspection_order_id: int
    order_no: str
    target_status: InspectionStatus
    opinion: Optional[str] = None
    signature: Optional[str] = None
    current_version: Optional[int] = None


class BatchProcessRequest(BaseModel):
    items: List[BatchProcessItem]
    request_id: Optional[str] = None
    operation: str


class BatchItemResult(BaseModel):
    inspection_order_id: int
    order_no: str
    success: bool
    error_code: Optional[str] = None
    error_message: Optional[str] = None
    previous_status: Optional[InspectionStatus] = None
    current_status: Optional[InspectionStatus] = None
    suggestion: Optional[str] = None
    next_step: Optional[str] = None


class BatchStatistics(BaseModel):
    total_count: int
    success_count: int
    failed_count: int
    skipped_count: int = 0


class BatchProcessResult(BaseModel):
    success: bool
    message: str
    statistics: BatchStatistics
    results: List[BatchItemResult]
    request_id: Optional[str] = None
    completed_at: datetime
